ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -184,7 +184,10 @@ data LowLevelRunNodeArgs m addrNTN addrNTC versionDataNTN versionDataNTC blk
    LowLevelRunNodeArgs {
 
       -- | How to manage the clean-shutdown marker on disk
-      llrnWithCheckedDB :: forall a. (LastShutDownWasClean -> m a) -> m a
+      llrnWithCheckedDB :: forall a. (   LastShutDownWasClean
```

Comment:
Yes probably this deserves a datatype. The TVar means whether it should create a new `clean` file in presence of a non-corrupting exception. The function is the one to remove the old `clean` file.

Reply:
Yeah, this comment was terse. We could elaborate it to, eg:

```
-- | A bracket function that manages the clean-shutdown marker on disk.
--
-- o If the marker is missing on startup, then ChainDB initialization will revalidate the database contents.
--
-- o If the OS kills the nodes, then we don't even have the opportunity to write out the marker file, which is appropriate, since we want the next startup to do revalidation. (TODO But need we revalidate _everything_ in this case?)
--
-- o If initialization was cleanly interrupted (eg SIGINT), then we leave the marker how we found it upon this starting of the node.
--
-- o At the end of a successful initialization, we remove the marker and install a shutdown handler that writes the marker except for certain exceptions (see 'exceptionRequiresRecovery') that suggest corruption, for which we want the next startup to do revalidation.
```

Reply:
Added this documentation